liblepton/scheme/lepton/object/type.scm

```diff
@@ -0,0 +1,185 @@
+               pointer
+               (scm-error 'wrong-type-arg
+                          proc-name
+                          "Wrong type argument in position ~A (expecting ~A object): ~A"
+                          (list pos type object)
+                          #f)))))))
+
+
+(define (object? object)
+  "Returns #t if OBJECT is a #<geda-object> instance, otherwise
+returns #f."
+  (true? (edascm_is_object (scm->pointer object))))
+
+
+(define (arc? object)
+  "Returns #t if OBJECT is a arc object, otherwise returns #f."
+  (true? (lepton_object_is_arc (geda-object->pointer object))))
+
+(define (box? object)
+  "Returns #t if OBJECT is a box object, otherwise returns #f."
+  (true? (lepton_object_is_box (geda-object->pointer object))))
+
+(define (bus? object)
+  "Returns #t if OBJECT is a bus object, otherwise returns #f."
+  (true? (lepton_object_is_bus (geda-object->pointer object))))
+
+(define (circle? object)
+  "Returns #t if OBJECT is a circle object, otherwise returns #f."
+  (true? (lepton_object_is_circle (geda-object->pointer object))))
+
+(define (component? object)
+  "Returns #t if OBJECT is a component object, otherwise returns #f."
+  (true? (lepton_object_is_component (geda-object->pointer object))))
+
+(define (line? object)
+  "Returns #t if OBJECT is a line object, otherwise returns #f."
+  (true? (lepton_object_is_line (geda-object->pointer object))))
+
+(define-public (net? object)
+  "Returns #t if OBJECT is a net object, otherwise returns #f."
+  (true? (lepton_object_is_net (geda-object->pointer object))))
+
+(define (path? object)
+  "Returns #t if OBJECT is a path object, otherwise returns #f."
+  (true? (lepton_object_is_path (geda-object->pointer object))))
+
+(define (picture? object)
+  "Returns #t if OBJECT is a picture object, otherwise returns #f."
+  (true? (lepton_object_is_picture (geda-object->pointer object))))
+
+(define (pin? object)
+  "Returns #t if OBJECT is a pin object, otherwise returns #f."
+  (true? (lepton_object_is_pin (geda-object->pointer object))))
+
+(define (text? object)
+  "Returns #t if OBJECT is a text object, otherwise returns #f."
+  (true? (lepton_object_is_text (geda-object->pointer object))))
+
+
+(define (object-type object)
+  "Returns a Scheme symbol representing the type of OBJECT.  The
+type may be one of the symbols: 'arc, 'box, 'bus, 'circle,
+'complex, 'line, 'net, 'path, 'picture, 'pin, or 'text."
+  (define pointer (geda-object->pointer* object 1))
+
+  (cond
+   ((arc? object) 'arc)
+   ((box? object) 'box)
+   ((bus? object) 'bus)
+   ((circle? object) 'circle)
+   ((component? object) 'complex)
+   ((line? object) 'line)
+   ((net? object) 'net)
+   ((path? object) 'path)
+   ((picture? object) 'picture)
+   ((pin? object) 'pin)
+   ((text? object) 'text)
+   (else (error "Object ~A has bad type '~A'"
+                object
+                (integer->char (lepton_object_get_type pointer))))))
+
+(define (object-type? object type)
```

Comment:
From the name `object-type?` I would guess that the function
takes one symbol argument and checks if it's a valid type name.
`object-type-is?()`?

Reply:
Well, this was implemented long ago and was in the code and docs for years.  IMO, decisions on naming were not always TRT.  There are Scheme functions, for example, that have `object` as part of their names and those work with *any* Scheme objects.  More TLC is needed to develop a new naming scheme for lepton functions and gradually work on replacing and deprecation of old named ones.